### ISSUE TITLE
Overwriting STDIN with a JavaScript function.

### ISSUE THREAD

**Issue by adminy** [pypyjs/pypyjs#206]
I want to be able to say 
`window.vm = new pypyjs({`
`       totalMemory:  256 * 1024 * 1024,`
`       stdin: function() { return window.data_var; }`
` });`

But it dosen't work, it just trows this back.

> Traceback

> (most recent call last):
>  File \"<console\>", line 1, in <module>
>EOFError

**Comment by rfk** [comment]
@adminy to be honest I haven't had a lot of luck getting stdin to behave sensibly in general - python wants to ready from it synchronously, which is a bad fit for the javascript environment.  I'm happy to have a look though - are you able to post a complete example (e.g. command-line javscript file, or web page) that throws the above error for me to take a look at?

**Comment by adminy** [comment]
@rfk thank you, actually, I wasn't realising which pypyjs instance I was actually working with is why I was getting that weird error. Anyways I did get sys.stdin to work properly for me. I allow users to upload a file which will behave like sys.stdin. What I did is overwrite the meaning of sys.stdin with a open('file'). It works like a charm and It might actually be the solution to this problem. Here is a demo of it working: https://adminy.github.io/py/ type some code in the editor like
import sys
a = input()
print(a)
for line in sys.stdin:print(line)

choose the sys.stdin file and type run in the console. It will work. Also Typing 
a = input()
print(a)
in the console itself should also work!
